File: code/Java/interpreter/command/AssignCommand.java
package interpreter.command;

import interpreter.expr.IntExpr;
import interpreter.expr.Variable;

public class AssignCommand extends Command {

    private Variable var;
    private IntExpr expr;

    public AssignCommand(int line, Variable var, IntExpr expr) {
        super(line);
        this.var = var;
        this.expr = expr;
    }

    public void execute() {
        int value = expr.expr();
        var.setValue(value);
    }

}

File: code/C++/interpreter/command/AssignCommand.cpp
#include "AssignCommand.h"
#include "../expr/Variable.h"

AssignCommand::AssignCommand(int line, Variable* var, IntExpr* expr)
	: Command(line), m_var(var), m_expr(expr) {
}

AssignCommand::~AssignCommand() {
	delete m_var;
	delete m_expr;
}

void AssignCommand::execute() {
	int value = m_expr->expr();
	m_var->setValue(value);
}
